src/Compilers/CSharp/Portable/FlowAnalysis/PreciseAbstractFlowPass.cs

```diff
@@ -2768,6 +2768,52 @@ public override BoundNode VisitNonConstructorMethodBody(BoundNonConstructorMetho
             return null;
         }
 
+        public override BoundNode VisitNullCoalesingAssignmentOperator(BoundNullCoalesingAssignmentOperator node)
+        {
+            LocalState savedState;
+            if (RegularPropertyAccess(node.LeftOperand))
+            {
+                var left = (BoundPropertyAccess)node.LeftOperand;
+                var property = left.PropertySymbol;
+                if (property.RefKind == RefKind.None)
+                {
+                    var readMethod = property.GetOwnOrInheritedGetMethod() ?? property.SetMethod;
+                    var writeMethod = property.GetOwnOrInheritedSetMethod() ?? property.GetMethod;
+
+                    Debug.Assert(node.HasAnyErrors || (object)readMethod != (object)writeMethod);
+
+                    VisitReceiverBeforeCall(left.ReceiverOpt, readMethod);
+                    if (_trackExceptions) NotePossibleException(node);
+                    VisitReceiverAfterCall(left.ReceiverOpt, readMethod);
+
+                    savedState = this.State.Clone();
+
+                    VisitRvalue(node.RightOperand);
+                    PropertySetter(node, left.ReceiverOpt, writeMethod);
+
+                    ConditionallyAssignNullCoalescingOperator(node);
+
+                    IntersectWith(ref this.State, ref savedState);
+                    return null;
+                }
+            }
+
+            VisitRvalue(node.LeftOperand);
```

Comment:
Why VisitRvalue?

Reply:
Talked with @gafter. `VisitRvalue` will ensure definite assignment rules, and we can't double visit as that would end up double-visiting any nested expressions.

---
In reply to: [199978223](https://github.com/dotnet/roslyn/pull/28253#discussion_r199978223) [](ancestors = 199978223)